app/process_proposal.go

```diff
@@ -118,73 +118,70 @@ func (app *App) ProcessProposal(req abci.RequestProcessProposal) (resp abci.Resp
 		// - that the sizes match
 		// - that the namespaces match between blob and PFB
 		// - that the share commitment is correct
-		if err := blobtypes.ValidateBlobTx(app.txConfig, blobTx, subtreeRootThreshold, app.AppVersion()); err != nil {
-			logInvalidPropBlockError(app.Logger(), req.Header, fmt.Sprintf("invalid blob tx %d", idx), err)
-			return reject()
+		if err := blobtypes.ValidateBlobTx(app.encodingConfig.TxConfig, blobTx, subtreeRootThreshold, appVersion); err != nil {
+			logInvalidPropBlockError(app.Logger(), blockHeader, fmt.Sprintf("invalid blob tx %d", idx), err)
+			return reject(), nil
 		}
 
 		// validated the PFB signature
-		sdkCtx, err = handler(sdkCtx, sdkTx, false)
+		ctx, err = handler(ctx, sdkTx, false)
 		if err != nil {
-			logInvalidPropBlockError(app.Logger(), req.Header, "invalid PFB signature", err)
-			return reject()
+			logInvalidPropBlockError(app.Logger(), blockHeader, "invalid PFB signature", err)
+			return reject(), nil
 		}
 
 	}
 
-	var (
-		dataSquareBytes [][]byte
-		err             error
-	)
+	var dataSquareBytes [][]byte
 
-	switch app.AppVersion() {
-	case v3:
+	switch appVersion {
+	case v4, v3:
```

Comment:
same as questions above: could we get rid of this switch since the multiplexer is handling versioning?